contracts/tokenomics/generator/src/contract.rs

```diff
@@ -454,22 +430,53 @@ pub fn update_pool_rewards(
     Ok(())
 }
 
+fn receive_cw20(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    cw20_msg: Cw20ReceiveMsg,
+) -> Result<Response, ContractError> {
+    let sender = deps.api.addr_validate(cw20_msg.sender.as_str())?;
+    let amount = cw20_msg.amount;
+
+    match from_binary(&cw20_msg.msg)? {
+        Cw20HookMsg::Deposit { lp_token } => update_rewards_and_execute(
```

Comment:
i think we need to use `info.sender` instead of lp_token, because otherwise we create a possibility to send fake LP tokens but still put them to proper pool. Same vulnerability is on the DepositFor method